elf/io/extensions.py

```diff
@@ -1,31 +1,102 @@
-from .knossos_wrapper import KnossosFile
+import functools
+
+import numpy as np
+
+from .knossos_wrapper import KnossosFile, KnossosDataset
+
+
+__all__ = [
+    "FILE_CONSTRUCTORS", "GROUP_LIKE", "DATASET_LIKE",
+    "h5py", "z5py", "pyn5", "zarr", "zarr_open",
+]
 
-# Kudos to @clbarnes for the extension design
 FILE_CONSTRUCTORS = {}
+ZARR_EXTS = [".zarr", ".zr"]
+N5_EXTS = [".n5"]
+
+GROUP_LIKE = []
+DATASET_LIKE = [np.ndarray]
```

Comment:
What's the reasoning of putting the ndarray in here?
It supports the same API, so should be ok to add it, but what are the benefits you foresee having it pass as a dataset?

Reply:
Use case: I have a hacky debug tool for looking at small 3D image data in matplotlib, called smalldataviewer, which tries to abstract over lots of array-like things so it's flexible about what you pass it. I considered rebuilding it on top of napari, but realised would would just be doing heimdall but much worse :smile: Anyway, in that application it was convenient to pass just about anything array-like into the same constructor, in which case it was nice to recognise numpy arrays, plus imageio volumes and PIL images (both subclasses of ndarray, IIRC) as being API-equivalent.

Reply:
Ok, I am convinced ;).